### PR TITLE
Support for version and some small fixes

### PR DESCRIPTION
**Pull request by yhoogstrate**
Supports gsl.py --package_version

Moves gsl to cargoport because otherwise it may try to import from bin/gsl.py instead of the ./gsl directory

Fixes that the column `checksum` has been renamed to `sha256sum`

Uses log instead of print()

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ You can use the community package cache with a small script called `gsl` (get st
 The simplest way to download your archive is using this magic curl command.
 
 ```console
-$ curl https://raw.githubusercontent.com/galaxyproject/cargo-port/master/gsl.py | python - --package_id augustus_3_1
+$ git clone https://github.com/galaxyproject/cargo-port.git ; cd cargo-port ; python bin/gsl.py --package_id samtools --package_version "1.2"
```
Comment (erasche) [inline README.md]: ah sad day. Let's either add a post-processing step, or put this on pypi! (But that can be a separate issue, not requesting that of you :) )

File: bin/gsl.py [hunk 2]
```diff
@@ -6,21 +6,28 @@
 import os
 import hashlib
 import logging
-from gsl.utils import yield_packages, package_name, PACKAGE_SERVER, depot_url
+import gsllib
+from gsllib.utils import yield_packages, package_name, PACKAGE_SERVER, depot_url
```
Comment (erasche) [inline bin/gsl.py]: if we're renaming, what do you think of renaming the package to something cargo-port related?

File: bin/gsl.py [hunk 3]
```diff
@@ -6,21 +6,28 @@
...
 logging.basicConfig(level=logging.DEBUG)
 log = logging.getLogger()
 
 
 @click.command()
 @click.option('--package_id', help='Package ID', required=True)
-@click.option('--download_location', default='./',
+@click.option('--package_version', help="Package version, downloads all versions if not specified", default=None, required=False)
+@click.option('--download_location', default='./downloads',
```
Comment (erasche) [inline bin/gsl.py]: any reason for defaulting to downloads directory?

**Comment by erasche** [comment]
Travis seems to have failed with the following:

```
error: package directory 'gsl' does not exist
/home/travis/build.sh: line 45: galaxy-cachefile-validator: command not found
The command "python setup.py build ; python setup.py install ; galaxy-cachefile-validator urls.tsv" exited with 127.
```

File: bin/gsl.py (hunk 3 above)
Comment (yhoogstrate) [inline bin/gsl.py]: Yep, it was in the `.gitignore` :)

File: bin/gsl.py (hunk 2 above)
Comment (yhoogstrate) [inline bin/gsl.py]: This makes absolutely sense

File: bin/gsl.py (hunk 3 above)
Comment (erasche) [inline bin/gsl.py]: Ok, I don't use gsl.py at all, so maybe @bgruening is better to comment here

File: README.md (hunk 1 above)
Comment (erasche) [inline README.md]: can you also add to the main readme please? https://github.com/galaxyproject/cargo-port#using-tcp

**Comment by bgruening** [comment]
:+1: Thanks @yhoogstrate!

**Pull request merged**